samples/java-protobuf-valueentity-shopping-cart/src/test/java/com/example/shoppingcart/ShoppingCartActionImplTest.java

```diff
@@ -47,7 +48,7 @@ public void tearDown() throws Exception {
   // tag::initialize[]
   @Test
   public void initializeCartTest() throws ExecutionException, InterruptedException, TimeoutException {
-    when(shoppingCartService.create(notNull()))
+    when(shoppingCartService.create().invoke(notNull()))
```

Comment:
Under the hood we have:
```
(Metadata metadata) -> addHeaders(((com.example.shoppingcart.ShoppingCartServiceClient) getGrpcClient(com.example.shoppingcart.ShoppingCartService.class)).create(), metadata).invoke(createCart)
```
thanks to `RETURNS_DEEP_STUBS` I was able to fix it in the java version, but with mockito scala I'm not sure if this is an option. 

Maybe we shouldn't encourage users to write unit tests for side effects. Personally I would go with IT tests for such cases.